spring-integration-core/src/test/java/org/springframework/integration/transformer/ProtoTests.java

```diff
@@ -0,0 +1,263 @@
+import org.springframework.core.log.LogAccessor;
+import org.springframework.integration.channel.DirectChannel;
+import org.springframework.integration.channel.QueueChannel;
+import org.springframework.integration.config.EnableIntegration;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.test.condition.LogLevels;
+import org.springframework.integration.test.util.TestUtils;
+import org.springframework.integration.transformer.proto.TestClass1;
+import org.springframework.integration.transformer.proto.TestClass2;
+import org.springframework.integration.transformer.support.ProtoHeaders;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.messaging.PollableChannel;
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+/**
+ *
+ * @author Christian Tzolov
+ * @since 6.1
+ */
+@SpringJUnitConfig
+public class ProtoTests {
+
+        @Test
+        @LogLevels(classes = DirectChannel.class, categories = "bar", level = "DEBUG")
+        void testTransformers(@Autowired ProtoConfig config) {
+                TestClass1 test = TestClass1.newBuilder()
+                                .setBar("foo")
+                                .setQux(678)
+                                .build();
+                LogAccessor spied = spy(TestUtils.getPropertyValue(config.in1(), "logger", LogAccessor.class));
+                new DirectFieldAccessor(config.in1()).setPropertyValue("logger", spied);
+                config.in1().send(new GenericMessage<>(test));
+                assertThat(config.tapped().receive(0))
+                                .isNotNull()
+                                .extracting(msg -> msg.getPayload())
+                                .isInstanceOf(byte[].class);
+                Message<?> received = config.out().receive(0);
+                assertThat(received)
+                                .isNotNull()
+                                .extracting(msg -> msg.getPayload())
+                                .isEqualTo(test)
+                                .isNotSameAs(test);
+                assertThat(received.getHeaders().get("flow")).isEqualTo("flow1");
+                ArgumentCaptor<String> captor = ArgumentCaptor.forClass(String.class);
+                verify(spied, atLeastOnce()).debug(captor.capture());
+                assertThat(captor.getAllValues()).anyMatch(s -> s.contains("preSend on channel"));
+                assertThat(captor.getAllValues()).anyMatch(s -> s.contains("postSend (sent=true) on channel"));
+        }
+
+        @Test
+        @LogLevels(classes = DirectChannel.class, categories = "bar", level = "DEBUG")
```

Comment:
I still see these on method.
What do you have in mind, please, to keep them?

Reply:
I've borrowed this form the Simple Avro Transfomer and apparently it is requred for the `LogAccessor spied` logic and verification to work. 
Apparently it uses reflection to substitute the parent IntegrationObjectSupport#logger with a spy instance and then verifies that the loger `debug` is beaing used at lest once. Do you know why this is important? otherwise i will remove all this logger tests.

Reply:
OK. I see what you are talking about the spying on the `logger`.
And why do we need to do that in this our tests?
Isn't that enough to be sure that we receive from transformers whatever we expect?
Why do we need to verify those `preSend on channel` and `postSend (sent=true) on channel`?
How is that logic relevant to Proto?